smalltalksrc/VMMaker/SpurMemoryManager.class.st

```diff
@@ -10231,6 +10288,20 @@ SpurMemoryManager >> outerContextOf: aBlockClosureOop [
 	^ self fetchPointer: FullClosureOuterContextIndex ofObject: aBlockClosureOop
 ]
 
+{ #category : 'interpreter access' }
+SpurMemoryManager >> paddingForFormatInstBytes: instBytes forElementBytes: elementSizeInBytes shiftForElement: shiftForElement [
+
+	"This method is doing the same as below but using bit arithmetic.
+	shiftForElement SHOULD be log 2 of element size in bytes. We don't do the log2 because 1) is slow 2) it can return floats
+	
+	slotsPerWord := (wordSize / elementSize).
+	trailingBytes := instBytes \\ wordSize.
+	trailingSlots := trailingBytes / elementSize.
+	^ slotsPerWord - trailingSlots. "
+
+	^ self wordSize // elementSizeInBytes - (instBytes >> shiftForElement) bitAnd: self wordSize // elementSizeInBytes - 1
+]
```

Comment:
I think we can do better:
```suggestion
{ #category : 'interpreter access' }
SpurMemoryManager >> paddingForFormatInstBytes: instBytes shiftForElement: shiftForElement [
	"NOTE: shiftForElement is elementSizeInBytes log: 2.
	This means elementSizeInBytes should be a power of two."
	| trailingBytes paddingBytes |
	"Cheap way to do instBytes \\ self wordSize (assumes self wordSize is a power of two)"
	trailingBytes := instBytes bitAnd: self wordSize - 1.
	paddingBytes := self wordSize - trailingBytes.
	"self wordSize - 0 is self wordSize, but we want to say 'there is no padding' in that case"
	paddingBytes := paddingBytes bitAnd: self wordSize - 1.
	"Cheap way to do paddingBytes // elementSizeInBytes"
	paddingBytes >> shiftForElement].
]
```

Reply:
NOTE: this change removes the elementBytes parameter :P

Reply:
Solved in [70346ca](https://github.com/pharo-project/pharo-vm/pull/857/commits/70346ca9c83f7df48043b2faf10095796a4ea2df)